version: "3.8"
services:
  netdata:
    image: netdata/netdata:latest    
    network_mode: host
    pid: host
    cap_add:
      - SYS_PTRACE
      - SYS_ADMIN
    security_opt:
      - apparmor:unconfined
    deploy:
      mode: global
      restart_policy:
        condition: on-failure
    labels:
      - "traefik.enable=true"
      - "traefik.docker.network=traefik_public_network"
      - "traefik.http.routers.netdata.rule=Host(`netdata.labz.tr`)"
      - "traefik.http.routers.netdata.entrypoints=websecure"
      - "traefik.http.routers.netdata.tls.certresolver=myresolver"
      - "traefik.http.services.netdata.loadbalancer.server.port=19999"
    networks:
      - traefik_public_network
    volumes:
      - netdata_config_data:/etc/netdata
      - netdata_lib_data:/var/lib/netdata
      - netdata_cache_data:/var/cache/netdata
      - /:/host/root:ro,rslave
      - /etc/passwd:/host/etc/passwd:ro
      - /etc/group:/host/etc/group:ro
      - /etc/localtime:/etc/localtime:ro
      - /proc:/host/proc:ro
      - /sys:/host/sys:ro
      - /etc/os-release:/host/etc/os-release:ro
      - /var/log:/host/var/log:ro
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - /etc/hostname:/host/etc/hostname:ro
networks:
    traefik_public_network:
      external: true
volumes:
  netdata_config_data:
  netdata_lib_data:
  netdata_cache_data: